themes/bootstrap5/js/facets.js

```diff
@@ -0,0 +1,301 @@
+      updateContent({facetpage: 1});
+    });
+  }
+
+  function setup() {
+    if ($.isReady) {
+      registerCallbacks();
+    } else {
+      $(function ready() {
+        registerCallbacks();
+      });
+    }
+  }
+
+  return { setup: setup, getContent: getContent, updateContent: updateContent };
+});
+
+/* --- Side Facets --- */
+VuFind.register('sideFacets', function SideFacets() {
+  function showLoadingOverlay() {
+    var overlay = '<div class="facet-loading-overlay">'
+      + '<span class="facet-loading-overlay-label">'
+      + VuFind.loading()
+      + "</span></div>";
+    $(this).closest(".collapse").append(overlay);
+  }
+
+  function activateFacetBlocking(context) {
+    var finalContext = (typeof context === "undefined") ? $(document.body) : context;
+    finalContext.find('a.facet:not(.narrow-toggle),.facet a').click(showLoadingOverlay);
+  }
+
+  function activateSingleAjaxFacetContainer() {
+    var $container = $(this);
+    var facetList = [];
+    var $facets = $container.find('div.collapse.in[data-facet], .checkbox-filter[data-facet]');
+    $facets.each(function addFacet() {
+      if (!$(this).data('loaded')) {
+        facetList.push($(this).data('facet'));
+      }
+    });
+    if (facetList.length === 0) {
+      return;
+    }
+    var request = {
+      method: 'getSideFacets',
+      searchClassId: $container.data('searchClassId'),
+      location: $container.data('location'),
+      configIndex: $container.data('configIndex'),
+      query: $container.data('query'),
+      querySuppressed: $container.data('querySuppressed'),
+      extraFields: $container.data('extraFields'),
+      enabledFacets: facetList
+    };
+    $container.find('.facet-load-indicator').removeClass('hidden');
+    $.getJSON(VuFind.path + '/AJAX/JSON?' + request.query, request)
+      .done(function onGetSideFacetsDone(response) {
+        $.each(response.data.facets, function initFacet(facet, facetData) {
+          var containerSelector = typeof facetData.checkboxCount !== 'undefined'
+            ? '.checkbox-filter' : ':not(.checkbox-filter)';
+          var $facetContainer = $container.find(containerSelector + '[data-facet="' + facet + '"]');
+          $facetContainer.data('loaded', 'true');
+          if (typeof facetData.checkboxCount !== 'undefined') {
+            if (facetData.checkboxCount !== null) {
+              $facetContainer.find('.avail-count').text(
+                facetData.checkboxCount.toString().replace(/\B(?=(\d{3})+\b)/g, VuFind.translate('number_thousands_separator'))
+              );
+            }
+          } else if (typeof facetData.html !== 'undefined') {
+            $facetContainer.html(VuFind.updateCspNonce(facetData.html));
+            activateFacetBlocking($facetContainer);
+          }
+          $facetContainer.find('.facet-load-indicator').remove();
+        });
+        VuFind.lightbox.bind($('.sidebar'));
+        VuFind.emit('VuFind.sidefacets.loaded');
```

Comment:
A favor for me: this shouldn't have `VuFind.` at the front.

Reply:
Sorry, I'm not following. This file is a direct copy from themes/bootstrap3/js/facets.js apart from the session storage handling.

Reply:
Ok. :) 

This reminds me that there's a slight problem with events in the current facet lightbox code: the modal resize relies on the fact that Bootstrap 3 triggers the show.bs.modal event on each show command even when the modal is already visible, while Bootstrap 5 doesn't. I'll post a fix for that separately after this PR has been merged.